src/RequestHandler.php

```diff
@@ -30,7 +30,7 @@ class RequestHandler implements MiddlewareInterface
      */
     public function __construct(ContainerInterface $container = null)
     {
-        $this->container = $container;
+        $this->container = $container ?: new RequestHandlerContainer();
```

Comment:
Because this is used only inside `if (is_string($requestHandler))`, I'd do not instantiate in the constructor, but inside that `if`.

Reply:
I think the consistency is valuable. There's no real performance impact of doing this.